InterviewFlashCards/JsonFileReader.swift

```diff
@@ -22,14 +22,23 @@ struct JsonFileReader: DataGenerator {
         guard let path = fileName.resourcePath,
             let jsonData = try? Data(contentsOf: path,
                                      options: .mappedIfSafe),
-            let json = (try? JSONSerialization
-                .jsonObject(with: jsonData as Data,
-                            options: .allowFragments)) as? [String: Any],
-            let data = json["data"] as? JSON else {
+            let json = json(from: jsonData),
+            let data = json["data"] as? [JSON] else {
                 completion([])
```

Comment:
Hey @charleshkang , so the `[JSON]`array here represents an array of dictionaries. `JSON` = `[String:Any]`, so `[JSON]` = `[ [String:Any] ]`. It was defined as a `typealias` in the QueryManager class, and since @vhart changed the algorithm questions text file to have a "data" key we only want to pass that into the completion block in the scenario that there's valid data from the file path, otherwise return an empty array...probably in cases that the file path is incorrect or something. Makes sense?